File: app/src/main/java/com/sachtech/datingapp/ui/login/fragment/SignUpFragment.java
package com.sachtech.datingapp.ui.login.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.google.firebase.auth.AuthResult;
import com.sachtech.datingapp.R;
import com.sachtech.datingapp.R.id;
import com.sachtech.datingapp.base.BaseFragment;
import com.sachtech.datingapp.extensions.CommonsKt;
import com.sachtech.datingapp.presenter.SignUpPresenter;
import com.sachtech.datingapp.utils.CommonUtil;
import com.sachtech.datingapp.utils.Commons;
import com.sachtech.datingapp.utils.IntentString;
import com.sachtech.datingapp.utils.PrefKeys;
import com.sachtech.datingapp.utils.Preferences;
import com.sachtech.datingapp.view.SignUpView;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public final class SignUpFragment extends Fragment implements SignUpView {


   public void onFailure(@NotNull String message) {
      new Commons(getContext()).hideProgress();
      CommonsKt.showToast(message);
   }

   public void onSignUpSuccess(@Nullable AuthResult result) {
      new Commons(getContext()).hideProgress();
      if(result!=null)
      {
         Preferences.INSTANCE.setPref(PrefKeys.INSTANCE.getPASSWORD(), signUpPassword.getText().toString());
         Preferences.INSTANCE.setPref(PrefKeys.INSTANCE.getUID(), result.getUser().getUid().toLowerCase());
         Preferences.INSTANCE.setPref(PrefKeys.INSTANCE.getUSERNAME(),signUpName.getText().toString());
         Preferences.INSTANCE.setPref(PrefKeys.INSTANCE.getEMAIL(),result.getUser().getEmail());
         bundle.putString(IntentString.INSTANCE.getEMAIL(), result.getUser().getEmail());
        bundle.putString(IntentString.INSTANCE.getNAME(), signUpName.getText().toString());
        BaseFragment.replaceFragment((AppCompatActivity) getActivity(),id.login_container,new ProfileImageFragment(),bundle);
      }
   }

   @NotNull
   public final SignUpPresenter getSignUpPresenter() {

      return new SignUpPresenter(this);
   }

   @androidx.annotation.Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @androidx.annotation.Nullable ViewGroup container, @androidx.annotation.Nullable Bundle savedInstanceState) {
      return LayoutInflater.from(getContext()).inflate(R.layout.fragment_sign_up,container,false);
   }
   Bundle bundle;
Button signUpBtn;
   TextView email,password,name,confirmPassword,signUpSignIn;
   EditText signUpEmail,signUpPassword,signUpName,signUpConfirmPassword;
   public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
      super.onViewCreated(view, savedInstanceState);
      CommonUtil instance = CommonUtil.INSTANCE;
       instance.hideKeyboard(getActivity());
       bundle=new Bundle();
      //setSpan();
     //
      signUpBtn=view.findViewById(id.signUpBtn);
      email=view.findViewById(id.email_header);
      password=view.findViewById(id.password);
      name=view.findViewById(id.name);
      confirmPassword=view.findViewById(id.confirmPassword);
      signUpEmail=view.findViewById(id.signUpEmail);
      signUpPassword=view.findViewById(id.signUpPassword);
      signUpName=view.findViewById(id.signUpName);
      signUpSignIn=view.findViewById(id.signupSignIn);
      signUpConfirmPassword=view.findViewById(id.signUpConfirmPassword);
       setOnFocusChange();
      signUpBtn.setOnClickListener(new OnClickListener() {
         @Override
         public void onClick(View v) {
            validateFields();
         }
      });
      signUpSignIn.setOnClickListener(new OnClickListener() {
          @Override
          public void onClick(View v) {
              BaseFragment.replaceFragment((AppCompatActivity) getActivity(),id.login_container,new SignInFragment(),null);

          }
      });

   }

   private  void setOnFocusChange() {
      CommonsKt.onFocusChange(signUpEmail, (View)email, "Email");
      CommonsKt.onFocusChange(signUpPassword, (View)password, "Password");
      CommonsKt.onFocusChange(signUpName, (View)name, "Name");
      CommonsKt.onFocusChange(signUpConfirmPassword, (View)confirmPassword, "Confirm Password");
   }

  /* private  void setSpan() {
      String span = signUpSignIn.getText().toString();
      SpannableString spannableString = new SpannableString((CharSequence)span);
      spannableString.setSpan(new ClickableSpan() {
         public void onClick(@NotNull View widget) {
            BaseFragment.replaceFragment((AppCompatActivity) getActivity(),id.login_container,new SignInFragment(),null);
         }
      }, 17, 25, 33);
      signUpSignIn.setText((CharSequence)spannableString);
    //  signUpSignIn.setMovementMethod(LinkMovementMethod.getInstance());
   }*/

   private  void validateFields() {
      if (!CommonsKt.validateEmail(signUpEmail.getText().toString())) {
         CommonsKt.showToast("Enter valid email");
      }

      if (!signUpPassword.getText().toString().equals(signUpConfirmPassword.getText().toString())) {
         CommonsKt.showToast("Password not matched");
      }

      CharSequence var5 = (CharSequence)signUpName.getText();
      boolean var6 = false;
      boolean var7 = false;
      if (var5 == null || var5.length() == 0) {
         CommonsKt.showToast("Name is required");
      }

      if (CommonsKt.validateEmail(signUpEmail.getText().toString()) && signUpPassword.getText().toString().equals(signUpConfirmPassword.getText().toString())) {
         var5 = (CharSequence)signUpName.getText();
         if (var5.length() > 0) {
             SignUpPresenter presenter=new SignUpPresenter(this);
            new Commons(getContext()).showProgress();
            presenter.signUpUser(signUpEmail.getText().toString(), signUpPassword.getText().toString());
         }
      }

   }
}

File: app/src/main/java/com/sachtech/datingapp/ui/login/fragment/SignUpFragment.kt
/*
package com.sachtech.datingapp.ui.login.fragment

import android.os.Bundle
import android.text.SpannableString
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.view.View
import com.google.firebase.auth.AuthResult
import com.sachtech.datingapp.R
import com.sachtech.datingapp.extensions.isNotNull
import com.sachtech.datingapp.extensions.onFocusChange
import com.sachtech.datingapp.extensions.showToast
import com.sachtech.datingapp.extensions.validateEmail
import com.sachtech.datingapp.presenter.SignUpPresenter
import com.sachtech.datingapp.utils.*
import com.sachtech.datingapp.utils.PreferencesKt.setPref
import com.sachtech.datingapp.view.SignUpView
import gurtek.mrgurtekbase.KotlinBaseFragment
import gurtek.mrgurtekbase.hideKeyboard
import kotlinx.android.synthetic.main.fragment_sign_up.*

class SignUpFragment : KotlinBaseFragment(R.layout.fragment_sign_up), SignUpView {

    override fun onFailure(message: String) {
        hideProgress()
        showToast(message)
    }

    override fun onSignUpSuccess(result: AuthResult?) {
        hideProgress()
        if (result.isNotNull()) {
            setPref(PrefKeys.INSTANCE.password, signUpPassword.text.toString())
            setPref(PrefKeys.UID, result?.user?.uid?.toLowerCase())
            setPref(PrefKeys.USERNAME,signUpName.text.toString())
            setPref(PrefKeys.EMAIL,result?.user?.email)
            arguments?.putString(IntentString.EMAIL, result?.user?.email)
            arguments?.putString(IntentString.NAME, signUpName.text.toString())
            baselistener.navigateToFragment(ProfileImageFragment::class, arguments)
        }

    }


    val signUpPresenter by lazy { SignUpPresenter(this) }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        CommonUtil.hideKeyboard(activity!!)

        setSpan()
        //setOnFocusChange()
        signUpBtn.setOnClickListener {
            validateFields()
        }
    }

    private fun setOnFocusChange() {
        onFocusChange(signUpEmail, email_header, "Email")
        onFocusChange(signUpPassword, password, "Password")
        onFocusChange(signUpName, name, "Name")
        onFocusChange(signUpConfirmPassword, confirmPassword, "Confirm Password")
    }

    private fun setSpan() {
        val span = signupSignIn.text.toString()
        val spannableString = SpannableString(span)
        spannableString.setSpan(object : ClickableSpan() {
            override fun onClick(widget: View) {
                baselistener.navigateToFragment(SignInFragment::class)
            }

        }, 17, 25, SpannableString.SPAN_EXCLUSIVE_EXCLUSIVE)
        signupSignIn.text = spannableString
        signupSignIn.movementMethod = LinkMovementMethod.getInstance()
    }

    private fun validateFields() {
        val email = signUpEmail.text.toString()
        val password = signUpPassword.text.toString()
        val confirmPassword = signUpConfirmPassword.text.toString()
        val name = signUpName.text.toString()
        if(!email.validateEmail())
        {
            showToast("Enter valid email")
        }
        if(password!=confirmPassword)
        {
            showToast("Password not matched")
        }
        if(name.isNullOrEmpty())
        {
            showToast("Name is required")
        }
        if (email.validateEmail() && password == confirmPassword && name.isNotEmpty()) {
            showProgress()
            signUpPresenter.signUpUser(email, password)
        }
    }
}*/
